src/core/border_router/routing_manager.cpp

```diff
@@ -0,0 +1,700 @@
+                                         const uint8_t *     aBuffer,
+                                         uint16_t            aBufferLength)
+{
+    OT_UNUSED_VARIABLE(aSrcAddress);
+    OT_UNUSED_VARIABLE(aBuffer);
+    OT_UNUSED_VARIABLE(aBufferLength);
+
+    otLogInfoBr("received Router Solicitation from %s on interface %s", aSrcAddress.ToString().AsCString(),
+                mInfraIfName);
+    SendRouterAdvertisement(mAdvertisedOmrPrefix, mAdvertisedOnLinkPrefix);
+}
+
+bool RoutingManager::HandlePrefixInfoOption(const RouterAdv::PrefixInfoOption &aPio)
+{
+    bool      needReevaluate = false;
+    TimeMilli expireTime;
+
+    if (aPio.GetValidLifetime() == 0)
+    {
+        if (mDiscoveredOnLinkPrefix.IsEqual(aPio.GetPrefix(), aPio.GetPrefixLength()))
+        {
+            otLogInfoBr("invalidate discovered on-link prefix: %s", mDiscoveredOnLinkPrefix.ToString().AsCString());
+            mDiscoveredOnLinkPrefix.Clear();
+            mDiscoveredOnLinkPrefixInvalidTimer.Stop();
+            SendRouterSolicit();
+        }
+        ExitNow();
+    }
+
+    expireTime = GetPrefixExpireTime(aPio.GetValidLifetime());
+
+    if (!IsValidOnLinkPrefix(mDiscoveredOnLinkPrefix) ||
+        (mDiscoveredOnLinkPrefixInvalidTimer.IsRunning() &&
+         expireTime > mDiscoveredOnLinkPrefixInvalidTimer.GetFireTime()))
+    {
+        mDiscoveredOnLinkPrefix.Set(aPio.GetPrefix(), aPio.GetPrefixLength());
+
+        otLogInfoBr("set discovered on-link prefix to %s, valid lifetime: %u seconds",
+                    mDiscoveredOnLinkPrefix.ToString().AsCString(), aPio.GetValidLifetime());
+
+        if (aPio.GetValidLifetime() == RouterAdv::kInfiniteLifetime)
+        {
+            mDiscoveredOnLinkPrefixInvalidTimer.Stop();
+        }
+        else
+        {
+            mDiscoveredOnLinkPrefixInvalidTimer.FireAt(expireTime);
+        }
+
+        mRouterSolicitTimer.Stop();
+        needReevaluate = true;
+    }
+
+exit:
+    return needReevaluate;
+}
+
+TimeMilli RoutingManager::GetPrefixExpireTime(uint32_t aValidLifetime)
+{
+    if (aValidLifetime * 1000ULL + TimerMilli::GetNow().GetValue() < TimeMilli::kMaxDuration)
+    {
+        return Time(aValidLifetime * 1000 + TimerMilli::GetNow().GetValue());
+    }
+    else
+    {
+        return Time(Time::kMaxDuration);
+    }
+}
+
+void RoutingManager::HandleRouterAdvertisement(const Ip6::Address &aSrcAddress,
+                                               const uint8_t *     aBuffer,
+                                               uint16_t            aBufferLength)
+{
+    using RouterAdv::Option;
+    using RouterAdv::PrefixInfoOption;
+    using RouterAdv::RouterAdvMessage;
+
+    bool           needReevaluate = false;
+    const uint8_t *optionsBegin;
+    uint16_t       optionsLength;
+    const Option * option;
+
+    VerifyOrExit(aBufferLength >= sizeof(RouterAdvMessage));
+
+    OT_UNUSED_VARIABLE(aSrcAddress);
```

Comment:
move OT_UNUSED_VARIABLE to the beginning of method according to our style guide.